src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp

```diff
@@ -0,0 +1,212 @@
+            bool valid = (flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr));
+            if (!valid)
+            {
+                std::cerr << "Couldn't load schema file: " << FLATBUFFER_PATH << std::endl;
+            }
+            else
+            {
+                const std::string payload = data.at("payload").dump();
+                const std::string resource = data.at("resource").get<std::string>();
+                flatbuffers::Parser parser;
+
+                valid =
+                    (parser.Parse(flatbufferSchemaStr.c_str(), INCLUDE_DIRECTORIES) && parser.Parse(payload.c_str()));
+                if (!valid)
+                {
+                    std::cerr << "Error: " << parser.error_ << std::endl;
+                }
+                else
+                {
+                    uint8_t* buf = parser.builder_.GetBufferPointer();
+                    size_t flatbufferSize = parser.builder_.GetSize();
+
+                    const rocksdb::Slice translationSlice(reinterpret_cast<const char*>(buf), flatbufferSize);
+                    rocksDbWrapper.put(resource, translationSlice);
+
+                    ret = true;
+                }
+            }
+        }
+        else
+        {
+            std::cerr << "No payload data" << std::endl;
+        }
+
+        return ret;
+    }
+
+    /**
+     * @brief Updates a translation entry in a rocksdb database.
+     *
+     * @param data JSON with the required information to create or update a translation entry.
+     * @param rocksDbWrapper rocksdb wrapper instance.
+     */
+    static bool updateTranslationEntry(const nlohmann::json& data, Utils::RocksDBWrapper& rocksDbWrapper)
+    {
+        std::string translationEntryStr;
+        bool ret = false;
+
+        if (rocksDbWrapper.get(data.at("resource").get<std::string>(), translationEntryStr))
+        {
+            const std::string resource = data.at("resource").get<std::string>();
+            std::string flatbufferSchemaStr;
+
+            if (flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr))
+            {
+                flatbuffers::Parser parser;
+                parser.opts.strict_json = true;
+
+                bool valid = parser.Parse(flatbufferSchemaStr.c_str(), INCLUDE_DIRECTORIES);
+                if (!valid)
+                {
+                    std::cerr << parser.error_ << std::endl;
+                }
+                else
+                {
+                    std::string translationEntry;
+                    if (!flatbuffers::GenText(parser, translationEntryStr.c_str(), &translationEntry))
+                    {
+                        try
+                        {
+                            nlohmann::json translationEntryJSON = nlohmann::json::parse(translationEntry);
+                            nlohmann::json translationEntryJSONPatched =
+                                translationEntryJSON.patch(data.at("operations"));
```

Comment:
Using `https://json.nlohmann.me/api/basic_json/patch_inplace/` we can avoid a copy.

Reply:
I get this error, i need to review the implementation 
![image](https://github.com/wazuh/wazuh/assets/13010397/f9e6431d-22bc-4661-bea6-8af4cfdaaa02)

Reply:
I don't know why it isn't working.
We can ignore the suggestion.